.buildkite/common.py

```diff
@@ -92,8 +92,13 @@ def pipeline_to_json(pipeline):
 def get_changed_files():
     """
     Get all files changed since `branch`
+    For branch other than `main`:
+        Use BUILDKITE_BRANCH for nightly tests
+        Use BUILDKITE_PULL_REQUEST_BASE_BRANCH for PR tests
     """
-    branch = os.environ.get("BUILDKITE_PULL_REQUEST_BASE_BRANCH", "main")
+    branch = os.environ.get(
+        "BUILDKITE_PULL_REQUEST_BASE_BRANCH", os.environ.get("BUILDKITE_BRANCH", "main")
```

Comment:
I think this will always evaluate to `""`. It should be sthg like:

```
  branch = os.environ.get("BUILDKITE_PULL_REQUEST_BASE_BRANCH") or os.environ.get("BUILDKITE_BRANCH") or "main"
```

Reply:
huh, why does it evaluate to `""`?

Reply:
If the value of the environment variable is an empty string (which is what BK does IIRC), that will be returned as an empty string. That helps differentiate the case when the env var is not present.

```python
% BLA="" ipython3
In [1]: os.environ.get("BLA")
Out[1]: ''
```

Reply:
ohhh, interesting! I didn't realize buildkite would set it to empty, instead of not setting it at all. Thanks for the explanation!